src/modules/service-tag.c

```diff
@@ -0,0 +1,93 @@
+/*
+ *   IRC - Internet Relay Chat, src/modules/bot-tag.c
+ *   (C) 2021 Syzop & The UnrealIRCd Team
+ *
+ *   See file AUTHORS in IRC package for additional names of
+ *   the programmers.
+ *
+ *   This program is free software; you can redistribute it and/or modify
+ *   it under the terms of the GNU General Public License as published by
+ *   the Free Software Foundation; either version 1, or (at your option)
+ *   any later version.
+ *
+ *   This program is distributed in the hope that it will be useful,
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *   GNU General Public License for more details.
+ *
+ *   You should have received a copy of the GNU General Public License
+ *   along with this program; if not, write to the Free Software
+ *   Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+ */
+
+/*
+	This gives services the appropriate tag so clients
+	can verify that the password they're sending is
+	actually going to a real service and not just
+	someone who's taken over the authentication nick
+	from a netsplit or something.
+*/
+
+#include "unrealircd.h"
+
+ModuleHeader MOD_HEADER
+  = {
+	"service-tag",
+	"5.0",
+	"service message tag",
+	"UnrealIRCd Team",
+	"unrealircd-5",
+	};
+
+int servicetag_mtag_is_ok(Client *client, char *name, char *value);
+void mtag_add_servicetag(Client *client, MessageTag *recv_mtags, MessageTag **mtag_list, char *signature);
+
+MOD_INIT()
+{
+	MessageTagHandlerInfo mtag;
+
+	MARK_AS_OFFICIAL_MODULE(modinfo);
+
+	memset(&mtag, 0, sizeof(mtag));
+	mtag.name = "unrealircd.org/service";
```

Comment:
Why not `inspircd.org/service`?

Reply:
Why inspircd? It is UnrealIRCd.
Just wondering?

Reply:
`inspircd.org/` is used as a unique namespace, it doesn't mean only InspIRCd can use it. As long as Unreal behaves like InspIRCd, there's no reason to use a different name.

Reply:
I mean, I only found out inspircd had it today because I asked after I had the idea, I didn't take the idea from there, I don't see why it shouldn't use the vendor name

Reply:
So client developers don't have to track all implementations of the same concept, while making sure they are not subtly different.

Reply:
I agree in a sense, but I think here, the concept may have already been implemented in another ircd, however, to me, the 'concept' is "/service" and I think the vendor should be described.

And client implementations, the client would most likely have tokenized the message tags (as I do when client-side scripting), it's very easy to tokenise a token to see if "/service" is part of the token, and describing the vendor to me seems like additional security should a client choose to pursue that.

Reply:
I agree with @progval, if you're implementing the same functionality as inspircd I'd much rather see you use the existing cap. An implementations using the vendored cap from a different vendor isn't unheard of.

The string matching with "/service" is unsafe, a different vendor could choose to give a completely different meaning to a tag containing this string.

Reply:
The vendor is [used as a namespace](https://ircv3.net/specs/extensions/message-tags#vendor-specific) to avoid conflicting names when two implementations use the same word for different concepts. (to avoid the same clusterfuck as conflicting modes, like `+q`).

If clients ignore the vendor part and only use the name after the slash, this nullifies the usefulness namespacing; we might as well not use vendored names at all.

Additionally, splitting on `/` is forbidden by the message-tags spec: ["Implementations MUST treat tag key names as case-sensitive opaque identifiers"](https://ircv3.net/specs/extensions/message-tags#format)

Reply:
Well why didn't you say ;D

Reply:
You forgot to update this one (like 52)

Reply:
fixed lol 👀👀